webapp/src/pages/offer/components/OfferComponent.js

```diff
@@ -0,0 +1,210 @@
+import React, {Component} from "react";
+import {withRouter} from "react-router";
+import TextArea from "react";
+import {Chart} from "react-google-charts";
+import { offerServices  } from "../../../services/offer/offer.service";
+import { profileService} from "../../../services/profilelist/profilelist.service";
+
+const DEFAULT_EVENT_ID = process.env.REACT_APP_DEFAULT_EVENT_ID || 1;
+
+class Offer extends Component {
+  
+  constructor(props) {
+        super(props);
+  
+        this.state = {
+          userProfile : [],
+          loading: true,
+          error: "",
+          offer_id: 1,
+          rejected_reason:"",
+          rejected:false,
+          accepted: false,
+          offerList: [],
+          category:""
+        };
+      }
+    
+
+      handleChange = field => {
+        return event => {
+          this.setState({
+            [field.name]: event.target.value
+          });
+        };
+      };
+
+      buttonSubmit(){
+         const {offer_id,accepted,rejected,rejected_reason} = this.state;
+   
+        offerServices.updateOfferList(offer_id, DEFAULT_EVENT_ID, accepted, rejected, rejected_reason)
+        .then(response=>{
+          if (response.msg === "succeeded") {
+            this.getofferList();
+            this.setState({
+              addedSucess: true,
+              conflict: false,
+              notFound: false
+            });
+          } else if (response.msg === "404") {
```

Comment:
You should check on the response.error

Please look at: ReviewForm.js as an example:

```
               reviewService
                    .submit(this.state.form.response.id, this.state.form.review_form.id, scores, shouldUpdate)
                    .then(response => {
                        if (response.error) {
                            this.setState({
                                error: response.error,
                                isSubmitting: false
                            });
                        }
                        else {
                            if (this.state.form.review_response) {
                                this.props.history.push(`/reviewHistory`)
                            }
                            else {
                                this.loadForm();
                            }
                        }
                    });
```